stdlib/LinearAlgebra/src/dense.jl

```diff
@@ -875,7 +875,9 @@ julia> sqrt(A)
 sqrt(::AbstractMatrix)
 
 function sqrt(A::AbstractMatrix{T}) where {T<:Union{Real,Complex}}
-    if ishermitian(A)
+    if isempty(A)
```

Comment:
`A` can be empty, but "non-square". Perhaps we should check upfront before we do anything? The rest of the function assumes squarity anyway.

```suggestion
    checksquare(A)
    if isempty(A)
```

Reply:
If you're not going to use `n` again you might as well do:
`A` can be empty, but "non-square". Perhaps we should check upfront before we do anything? The rest of the function assumes squareness anyway.

```suggestion
    if checksquare(A) == 0
```

Reply:
sounds good, I'll make these changes

Reply:
@stevengj I think @dkarrasch suggested going by the `n = checksquare(A)` because in that case if `A` is not square, a DimensionMismatch error will be thrown right away, rather than going through the whole `sqrt()` function. I am making changes accordingly.

Reply:
I think @stevengj's proposal will throw just as well.

Reply:
My bad. Maybe I did not save my file or maybe there was something wrong with Revise. I got different stacktraces for the two, I found it weird too.

Reply:
I have made the changes, and added a test as well.